### PR TITLE
Update documents.

### PR DESCRIPTION
**Pull request by trivialfis**
* Remove best_ntree_limit.
* Better doxygen output.

### CODE REVIEW

File: doc/prediction.rst [hunk 1]
```diff
@@ -67,6 +67,18 @@ the 3-class classification dataset, and want to use the first 2 iterations of tr
 prediction, you need to provide ``iteration_range=(0, 2)``.  Then the first :math:`2
 \times 3 \times 4` trees will be used in this prediction.
 
+**************
+Early Stopping
+**************
+
+When a model is trained with early stopping, there is an inconsistent behavior between
+native Python interface and sklearn/R interfaces.  By default on R and sklearn interfaces,
```
Comment (hcho3) [inline doc/prediction.rst:75]: Is it possible to change the default to `save_best=True` to eliminate this discrepancy outright?

File: doc/python/python_intro.rst [hunk 2]
```diff
@@ -183,7 +183,7 @@ Early stopping requires at least one set in ``evals``. If there's more than one,
 
 The model will train until the validation score stops improving. Validation error needs to decrease at least every ``early_stopping_rounds`` to continue training.
 
-If early stopping occurs, the model will have three additional fields: ``bst.best_score``, ``bst.best_iteration`` and ``bst.best_ntree_limit``. Note that :py:meth:`xgboost.train` will return a model from the last iteration, not the best one.
+If early stopping occurs, the model will have three additional fields: ``bst.best_score``, ``bst.best_iteration``.  Note that :py:meth:`xgboost.train` will return a model from the last iteration, not the best one.
```
Comment (RAMitchell) [inline doc/python/python_intro.rst:186]: two additional fields

File: doc/prediction.rst (hunk 1 above)
Comment (RAMitchell) [reply doc/prediction.rst:75]: Might be a good idea for consistency.

**Review by RAMitchell — APPROVED** [review]

**Comment by trivialfis** [comment]
@hcho3 @RAMitchell Yes we can make them consistent.  But that will be a breaking change.

File: doc/python/python_intro.rst (hunk 2 above)
Comment (trivialfis) [reply doc/python/python_intro.rst]: Thanks for catching it.  Fixed.

**Pull request merged**